### PR TITLE
Improve PR template

### PR DESCRIPTION
**Pull request by akeller**
## Description

Adjusts the PR template, which people may not bother even reading. So let's check to see who is paying attention 😜 

* Orders the urgency list from most to least urgent
* Removes compound sentences from PR checklist, as many may stop reading
* Adds more comments/context to help people understand why DevEx needs this information and not just because "Amara, the DRI of Docs, said so"
* Adds docs infrastructure as an optional review component, with clear direction to discuss in issue or Slack first (however, doesn't tag @pepopowitz, the DRI of Docs Infra, so he doesn't have to deal with every PR)

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,24 +1,23 @@
 ## Description
-
-<!-- This helps the reviewers by providing an overview of what to expect in the PR. Linking to an issue is preferred but not required. -->
-<!-- Add an assignee and use component: labels for good hygiene! -->
+<!-- Provide an overview of what to expect in the PR. -->
+<!-- Link to an associated epic when applicable. --> 
+<!-- Add an assignee and use `component:` or version labels for good hygiene! -->
```
Comment (christinaausley) [inline .github/pull_request_template.md:4]: YES ✋

**Review by christinaausley — APPROVED** [review]
This looks awesome. The only other thing I tend to need to check often are redirects and if files are added to the sidebar(s). Not sure if it would be appropriate to request checking those boxes also, but can always just revisit in a future iteration, as I need to check this in every PR whether a box is checked or not anyways 👍

**Review by pepopowitz — DISMISSED** [review]
Love the clarity here ❤️❤️❤️

**Comment by akeller** [comment]
> This looks awesome. The only other thing I tend to need to check often are redirects and if files are added to the sidebar(s). Not sure if it would be appropriate to request checking those boxes also, but can always just revisit in a future iteration, as I need to check this in every PR whether a box is checked or not anyways 👍

How would you phrase this? Do you want two checkboxes, one for new pages and one for removing pages?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
-- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
-- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
-- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.
+- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
+- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).
+
+<!-- Adding or removing pages requires extra steps.
+- [ ] I included my new page in the sidebar file(s).
+- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
+-->
```
Comment (akeller) [inline .github/pull_request_template.md:26]: @christinaausley what do you think about this?

Comment (christinaausley) [reply .github/pull_request_template.md:26]: Stellar 🌟

**Review by christinaausley — APPROVED** [review]

**Pull request merged**